### PR TITLE
2021 annual report for SIG instrumentation

### PR DESCRIPTION
**Pull request by logicalhan**
This is the annual report for SIG Instrumentation (currently a WIP).

### CODE REVIEW

File: sig-instrumentation/annual-report-2022.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+   - Meeting attendance
+       - Meeting attendance is ~10 each week
+       - Triage attendance is ~5 each week
+   - Enhancement velocity
+       - 7 Active enhancements over the year
+
+3. Does your [CONTRIBUTING.md] help **new** contributors engage with your group specifically by pointing
+   to activities or programs that provide useful context or allow easy participation?
+
+   -
```
Comment (ehashman) [inline sig-instrumentation/annual-report-2022.md:60]:
```suggestion
   - We don't have our own CONTRIBUTING.md as the overall one is suitable for our SIG. However, we have [filed an issue](https://github.com/kubernetes/community/issues/6462) to create one to better document how to get involved with our subprojects.
```

File: sig-instrumentation/annual-report-2022.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+5. Does the group have contributors from multiple companies/affiliations?
+
+   - Yes: Red Hat, Google, Intel, ...
+
+6. Are there ways end users/companies can contribute that they currently are not?
+   If one of those ways is more full time support, what would they work on and why?
+
+   - Yes
+   -
```
Comment (ehashman) [inline sig-instrumentation/annual-report-2022.md:75]:
```suggestion
SIG leads performed a [staffing and gap analysis](https://docs.google.com/document/d/1qeoP6i7GBTVJuJE1AGY5iU9dqmAOxrjqkfNQ2-rBeyI/edit#heading=h.849b7ydpl7ip) for the SIG's projects. We definitely can use more help, and need more part-time/full-time contributors.

### Summary

- **KTLO:** 4 leads at 5%, 2 maintainers at 10% for core, 9-10 maintainers at 5% for 5 subprojects (can have overlap between roles, but need a minimum of 3-4 part-time contributors at 25%)
- **Feature work:** needs significant ongoing additional investment, minimum of 2-3 FT devs or features will continue to slip

### Details

- **KTLO:** Requires two experienced part-time maintainers at 10% (e.g. 2x4h = 8h/wk) in addition to the SIG leadership (chairs/TLs @ minimum of 2h/wk)
  - Chairs/TLs currently perform the bulk of this work but even amongst the four of them, **do not** have 8h total weekly allocated
- **Feature work:** requires significantly more investment from development and review time. 
  - E.g. Structured logging initiative requires a minimum of 2 FT staff for the duration of feature development from beta -> GA
  - Additional metric stability classes and graduation: 1PT maintainer at 25% for the duration of feature development
  - SIG currently owns a number of KEPs stuck in alpha/beta due to lack of dev resources:
      - https://github.com/kubernetes/enhancements/issues/383 
      - https://github.com/kubernetes/enhancements/issues/1668
      - https://github.com/kubernetes/enhancements/issues/1753 
      - https://github.com/kubernetes/enhancements/issues/2305 

### Subprojects

- Subprojects are currently mature/stable and mainly have KTLO needs
- kube-state-metrics: 2-3 experienced maintainers at 5%, currently staffed
- Metrics-server: 2 experienced maintainers at 5%, currently staffed
- Custom-metrics-apiserver: 1 experienced maintainer at 5%, currently staffed
- Klog: 2 experienced maintainers at 5%, has **no current staffing**
- Prometheus-adapter: 2 experienced maintainers at 5%, has **no current staffing**
```

File: sig-instrumentation/annual-report-2022.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+In future, this will be generated from delta of sigs.yaml from $YYYY-01-01 to $YYYY-12-31
+Manually visible via `git diff HEAD@{$YYYY-01-01} HEAD@{$YYYY-12-31} -- $sig-id/README.md`
+-->
+
+New in 2021:
+- [instrumentation](https://github.com/kubernetes-sigs/instrumentation)
+- [instrumentation-addons](https://github.com/kubernetes-sigs/instrumentation-addons)
+
+Retired in 2021:
+- [heapster](https://github.com/kubernetes-retired/heapster)
```
Comment (ehashman) [inline sig-instrumentation/annual-report-2022.md:102]:
```suggestion
```

Heapster was retired in 2018, we can remove this

File: sig-instrumentation/annual-report-2022.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+## Working groups
+
+<!--
+In future, this will be generated from delta of sigs.yaml from $YYYY-01-01 to $YYYY-12-31
+Manually visible via `git diff HEAD@{$YYYY-01-01} HEAD@{$YYYY-12-31} -- $sig-id/README.md`
+-->
+
+New in 2021:
+- [WG Structured Logging](https://git.k8s.io/community/$wg-id/) ([$YYYY report](https://git.k8s.io/community/$wg-id/annual-report-$YYYY.md))
```
Comment (ehashman) [inline sig-instrumentation/annual-report-2022.md:124]: Need to update links

File: sig-instrumentation/annual-report-2022.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+Operational tasks in [sig-governance.md]:
+
+- [ ] [README.md] reviewed for accuracy and updated if needed
+- [ ] [CONTRIBUTING.md] reviewed for accuracy and updated if needed
+      (or created if missing and your contributor steps and experience are different or more
+      in-depth than the documentation listed in the general [contributor guide] and [devel] folder.)
+- [x] Subprojects list and linked OWNERS files in [sigs.yaml] reviewed for accuracy and updated if needed
+- [x] SIG leaders (chairs, tech leads, and subproject owners) in [sigs.yaml] are accurate and active, and updated if needed
+- [ ] Meeting notes and recordings for 2021 are linked from [README.md] and updated/uploaded if needed
```
Comment (ehashman) [inline sig-instrumentation/annual-report-2022.md:136]: I did the notes archive but I think Han needs to finish uploads

File: sig-instrumentation/annual-report-2022.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+New in 2021:
+- [WG Structured Logging](https://git.k8s.io/community/$wg-id/) ([$YYYY report](https://git.k8s.io/community/$wg-id/annual-report-$YYYY.md))
+
+## Operational
+
+Operational tasks in [sig-governance.md]:
+
+- [ ] [README.md] reviewed for accuracy and updated if needed
+- [ ] [CONTRIBUTING.md] reviewed for accuracy and updated if needed
```
Comment (ehashman) [inline sig-instrumentation/annual-report-2022.md:131]: Pending https://github.com/kubernetes/community/issues/6462

File: sig-instrumentation/annual-report-2022.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+-->
+
+New in 2021:
+- [WG Structured Logging](https://git.k8s.io/community/$wg-id/) ([$YYYY report](https://git.k8s.io/community/$wg-id/annual-report-$YYYY.md))
+
+## Operational
+
+Operational tasks in [sig-governance.md]:
+
+- [ ] [README.md] reviewed for accuracy and updated if needed
```
Comment (ehashman) [inline sig-instrumentation/annual-report-2022.md:130]: I owe a small update which needs to be approved by steering.

File: sig-instrumentation/annual-report-2022.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+3. KEP work in 2021 (1.x, 1.y, 1.z):
+
+<!--
+In future, this will be generated from kubernetes/enhancements kep.yaml files
+1. with SIG as owning-sig or in participating-sigs
+2. listing 1.x, 1.y, or 1.z in milestones or in latest-milestone
+-->
+
+   - Stable
+     - [1209 - Metrics Stability](https://git.k8s.io/enhancements/...) - 1.21
```
Comment (ehashman) [inline sig-instrumentation/annual-report-2022.md:25]: Need to fix links in this section

**Comment by ehashman** [comment]
/retitle [WIP] 2021 annual report for SIG instrumentation

Han can you move the file? It's technically the report for last year: https://github.com/kubernetes/community/blob/master/sig-instrumentation/annual-report-2021.md

File: sig-instrumentation/annual-report-2022.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+     - [2845 - Deprecate klog-specific flags in k8s components](https://git.k8s.io/community/$link/README.md) - 1.23
+   - Pre-alpha
+     - [2831 - Kubelet OpenTelemetry Tracing](https://git.k8s.io/community/$link/README.md) - alpha in 1.24
+
+## Project health
+
+1. What areas and/or subprojects does your group need the most help with?
+   Any areas with 2 or fewer OWNERs? (link to more details)
+
+   - [kubernetes-sigs/custom-metrics-apiserver](https://github.com/kubernetes-sigs/custom-metrics-apiserver/blob/master/OWNERS) has 1 active approver
```
Comment (dashpole) [inline sig-instrumentation/annual-report-2022.md:42]: Remove custom-metrics-apiserver and metrics-server from this list.  I mistakenly considered someone inactive when they have actually been active.

File: sig-instrumentation/annual-report-2022.md (hunk 2 above)
Comment (dashpole) [reply sig-instrumentation/annual-report-2022.md:75]: The number of maintainers listed here for subprojects differs from what I wrote above, and I believe this version is correct (devstats didn't have anything for s-urbaniak, but it looks like he has done some reviews recently).  I commented above to fix.

Comment (dgrisonnet) [reply sig-instrumentation/annual-report-2022.md:75]: This version wasn't based on devstat but rather on each project maintainer's feelings about the health of the project. Although some of them have been partially staffed this year (e.g prometheus-adapter), none of the maintainers have currently time to invest in it.

File: sig-instrumentation/annual-report-2022.md (hunk 9 above)
Comment (dgrisonnet) [reply sig-instrumentation/annual-report-2022.md:42]: Based on https://github.com/kubernetes/community/pull/6463/files/83a275c21cb7143055f26b7a813e1c7cd628503c#r806269581, would it make sense to add klog?

Comment (logicalhan) [reply sig-instrumentation/annual-report-2022.md:42]: I don't think so, seriathius and pohly are pretty active on it.

File: sig-instrumentation/annual-report-2022.md (hunk 7 above)
Comment (logicalhan) [reply sig-instrumentation/annual-report-2022.md:155]: is this done?

Comment (ehashman) [reply sig-instrumentation/annual-report-2022.md]: Yes: https://github.com/kubernetes/community/pull/6464

File: sig-instrumentation/annual-report-2021.md [hunk 10]
```diff
@@ -23,81 +22,121 @@ In future, this will be generated from kubernetes/enhancements kep.yaml files
...
 4. If your group has special training, requirements for reviewers/approvers, or processes beyond the general [contributor guide],
    does your [CONTRIBUTING.md] document those to help **existing** contributors grow throughout the [contributor ladder]?
 
-   -
+   - N/A, we don't have special training or requirements
 
 5. Does the group have contributors from multiple companies/affiliations?
 
-   -
+   - Yes: Red Hat, Google, Intel, ...
```
Comment (ehashman) [inline sig-instrumentation/annual-report-2021.md:69]:
```suggestion
   - Yes: Google, Red Hat, Sony, VMware, Intel, independent contributors, and more
```

File: sig-instrumentation/annual-report-2021.md [hunk 11]
```diff
@@ -126,18 +156,20 @@ Operational tasks in [sig-governance.md]:
 - [ ] [CONTRIBUTING.md] reviewed for accuracy and updated if needed
```
Comment (ehashman) [inline sig-instrumentation/annual-report-2021.md:156]:
```suggestion
- [x] [CONTRIBUTING.md] reviewed for accuracy and updated if needed
```

add x in box above too

**Review by ehashman — COMMENTED** [review]
Generally LGTM, couple small things

File: sig-instrumentation/annual-report-2021.md [hunk 12]
```diff
@@ -23,81 +22,121 @@ In future, this will be generated from kubernetes/enhancements kep.yaml files
...
 
 ## Project health
 
 1. What areas and/or subprojects does your group need the most help with?
    Any areas with 2 or fewer OWNERs? (link to more details)
 
-   -
-   -
-   -
+   - [kubernetes-sigs/custom-metrics-apiserver](https://github.com/kubernetes-sigs/custom-metrics-apiserver/blob/master/OWNERS) has 1 active approver
```
Comment (dashpole) [inline sig-instrumentation/annual-report-2021.md:42]: Custom metrics apiserver and metrics-server still need to be removed.  See https://github.com/kubernetes/community/pull/6463#pullrequestreview-883075545

File: sig-instrumentation/annual-report-2021.md [hunk 13]
```diff
@@ -23,81 +22,121 @@ In future, this will be generated from kubernetes/enhancements kep.yaml files
...
+   - [kubernetes-sigs/metrics-server](https://github.com/kubernetes-sigs/metrics-server/blob/master/OWNERS) has 1 active approver
```
Comment (dashpole) [inline sig-instrumentation/annual-report-2021.md:43]:
```suggestion
```
custom metrics apiserver and metrics server have 2 active approvers, and should be removed.

File: sig-instrumentation/annual-report-2021.md [hunk 14]
```diff
@@ -23,81 +22,119 @@ In future, this will be generated from kubernetes/enhancements kep.yaml files
...
+### Details
+
+- **KTLO:** Requires two experienced part-time maintainers at 10% (e.g. 2x4h = 8h/wk) in addition to the SIG leadership (chairs/TLs @ minimum of 2h/wk)
+  - Chairs/TLs currently perform the bulk of this work but even amongst the four of them, **do not** have 8h total weekly allocated
+- **Feature work:** requires significantly more investment from development and review time. 
+  - E.g. Structured logging initiative requires a minimum of 2 FT staff for the duration of feature development from beta -> GA
+  - Additional metric stability classes and graduation: 1PT maintainer at 25% for the duration of feature development
+  - SIG currently owns a number of KEPs stuck in alpha/beta due to lack of dev resources:
+      - https://github.com/kubernetes/enhancements/issues/383 
+      - https://github.com/kubernetes/enhancements/issues/1668
```
Comment (dashpole) [inline sig-instrumentation/annual-report-2021.md:88]: This is now closed.
```suggestion
```

**Pull request merged**